pykokkos/core/fusion/trace.py

```diff
@@ -189,6 +222,56 @@ def fuse(self, operations: List[TracerOperation], strategy: str) -> List[TracerO
 
         raise RuntimeError(f"Unrecognized fusion strategy '{strategy}'")
 
+    def is_safe_to_fuse(self, current: List[TracerOperation], current_views: Set[ViewType], current_safety_info: Dict[Tuple[int, int], Tuple[AccessIndex, AccessMode]], next: TracerOperation, next_views: Set[ViewType]) -> bool:
+        """
+        Check whether the next operation is safe to fuse with the
+        current operations
```

Comment:
this comment should document the conditions under which it would be safe to fuse